docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
One other point here: The API can't require that you pass in the peer's public key to get back an object which lets you query your public key. If you do, there's no way for either side to send their public key since they're both waiting to receive the public key from their peer. You need two steps. The first step creates the private key and lets you get the corresponding public key to put in the handshake, and the second step takes the received public key and returns the shared key.